### PR TITLE
[aievec] Add support for 0-rank xfer reads

### PR DESCRIPTION
**Pull request by jsetoain**
This patch solves issue [#584](https://github.com/Xilinx/mlir-aie/issues/584)

### CODE REVIEW

File: lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp [hunk 1]
```diff
@@ -1878,6 +1878,26 @@ static LogicalResult printOperation(CppEmitter &emitter,
...
+  // error out
+  if (!emitter.hasValueInScope(source))
+    return failure();
+
+  if (failed(emitter.emitAssignPrefix(*expandShapeOp)))
+    return failure();
+
+  raw_indented_ostream &os = emitter.ostream();
+
+  os << emitter.getOrCreateName(source);
```
Comment (jamestcl-amd) [inline lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp:1896]: For my own understanding. This op essentially is a no-op in the cpp? It only does variable assignment like "v18 = v17"?

Comment (jsetoain) [reply lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp:1896]: Indeed, it's just forwarding the value, and [that is the semantics of the operation itself](https://mlir.llvm.org/docs/Dialects/MemRef/#memrefexpand_shape-memrefexpandshapeop): _"The operation is limited to such reassociations, where a dimension is expanded into one or multiple contiguous dimensions. Such reassociations never require additional allocs or copies."_

A better option would be to replace all uses of the op with the source value and call it a day, so you don't give extra work to the C++ compiler, but this was a more immediate solution. I still might give it a go before landing it.

Comment (jamestcl-amd) [reply lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp:1896]: I see, that sounds reasonable to me. Hopefully, our C++ compiler can figure that out. If not, we can go for your better option later.

**Review by jamestcl-amd — APPROVED** [review]
Looks good to me. Thanks for fixing the issue promptly.

**Pull request merged**